sdk_lib/sdk_container_common.sh

```diff
@@ -17,40 +17,42 @@ function yell() {
 }
 # --
 
-# Prevent the sourcetree from being used by multiple containers with different
-#  SDK release versions by use of a version lock.
+# Prevent the sourcetree from being used by multiple containers at once.
 # The lock is self-cleaning and does not need to be removed in scripts calling
 # this function.
-function lock_version() {
-    local version="$1"
-    local owner="$2"
-    local lock_version=""
-    local lock_owner=""
-    local lock_owner_active=""
-
-    if ! mkdir sdk_container/.lock 2>/dev/null; then
-
-        if [ -f sdk_container/.lock/version ] ; then
-            lock_version="$(cat sdk_container/.lock/version)"
-        fi
-        if [ -f sdk_container/.lock/owner ] ; then
-            lock_owner="$(cat sdk_container/.lock/owner)"
-            lock_owner_active="$(docker ps --no-trunc --filter name="$lock_owner" --format '{{.Status}}')"
-        fi
-
-        if [ -z "$lock_owner_active" ] ; then
-            yell "Removing stale SDK lock (owner: '$lock_owner', version: '$lock_version')"
-            rm -f sdk_container/.lock/version sdk_container/.lock/owner
-        elif [ "$lock_version" != "$version" ] ; then
-            echo "ERROR: container workspace is locked." >&2
-            echo "This means that an active SDK container is using this workspace."  >&2
-            echo "(owner: $lock_owner, version: $lock_version)"  >&2
-            return 1
-        fi
+function lock_sourcetree() {
+    # acquire the lock
+    if mkdir sdk_container/.lock 2>/dev/null; then
+        return 0
+    fi
+
+    # acquisition failed, can we safely clear the lock?
+    if [ ! -f sdk_container/.lock/owner ] ; then
+        echo "ERROR: container workspace is locked (but no owner is set!)." >&2
+        echo "       This might mean that another container is currently starting." >&2
+        echo "       If you're SURE there's no other container using the sourcetree," >&2
+        echo "       run 'rm -rf sdk_container/.lock'. " >&2
+        return 1
+    fi
+
+    lock_owner="$(cat sdk_container/.lock/owner || true)"
+    lock_owner_active="$(docker ps --no-trunc \
+                              --filter name="$lock_owner" \
+                              --format '{{.Status}}')"
+
+    if [ -z "$lock_owner_active" ] ; then
+        yell "Removing stale SDK lock (owner: '$lock_owner')"
+        rm -rf sdk_container/.lock/
```

Comment:
nit: maybe add a comment here that this is not race free when executed in parallel (two executions want to clear the lock), because this `rm` line is able to remove the newly created lock from in case the other execution already deleted it and recreated it again

Reply:
Nice catch! Looks like I need to lock the owner check, too.